### PR TITLE
Create restaurant and cuisine type endpoints

### PR DESCRIPTION
**Pull request by RomanMykushLnu**
This commits:
1. Creates controllers, services, repositories for restaurants and cuisine types
2. Renames Utils layer to Application
3. Adds AutoMapper dependency
4. Slightly improves code formatting

### CODE REVIEW

File: API/RestMatch.API/RestMatch.API/Program.cs [hunk 1]
```diff
@@ -37,6 +51,14 @@
 app.UseOpenApi();
 app.UseSwaggerUi();
 
+if (app.Environment.IsDevelopment())
```
Comment (ikogarasi) [inline API/RestMatch.API/RestMatch.API/Program.cs]: move database seed into dbContext class

File: API/RestMatch.API/RestMatch.API/Program.cs [hunk 2]
```diff
@@ -14,6 +20,14 @@
 });
 
 builder.Services.AddControllers();
+
+builder.Services.AddAutoMapper(AppDomain.CurrentDomain.GetAssemblies());
+
+builder.Services.AddScoped<IRestaurantRepository, RestaurantRepository>();
```
Comment (ikogarasi) [inline API/RestMatch.API/RestMatch.API/Program.cs]: It's better to create separate extension methods for injecting repositories and services inside their layers and reuse here (see. injection of dbcontext)

File: API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        {
+            var cuisine = await _service.GetRestaurantCuisine(id);
+            if (cuisine == null)
+                return NotFound($"{nameof(RestaurantCuisine)} wasn't found");
+
+            return Ok(cuisine);
+        }
+
+        [HttpGet("{restaurantId}/cuisines")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
```
Comment (ikogarasi) [inline API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs]: add also type of response for nswag documentation like [ProducesResponseType(typeof(ICollection<GetRestaurantCuisineResponseDto>), StatusCodes.Status200OK)]

File: API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                return NotFound($"{nameof(RestaurantImageUrl)} wasn't found");
+
+            return Ok(imageUrl);
+        }
+
+        [HttpGet("{restaurantId}/imageUrls")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<ICollection<GetRestaurantImageUrlResponseDto>>> GetRestaurantImageUrls(int restaurantId)
```
Comment (ikogarasi) [inline API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs]: i suppose we don't need separate requests for restaurant, restaurant images and other stuff. The point was to retrieve it in one dto

File: API/RestMatch.API/RestMatch.API.Infrastructure/Repositories/RestaurantRepository.cs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        {
+            _context = context;
+        }
+
+        public async Task<ICollection<Restaurant>> GetRestaurants() =>
+            await _context.Restaurants.ToListAsync();
+
+        public async Task<Restaurant?> GetRestaurant(int id) =>
+            await _context.Restaurants.Where(r => r.Id == id)
+                .Include(r => r.Cuisines).Include(r => r.ImageUrls)
```
Comment (ikogarasi) [inline API/RestMatch.API/RestMatch.API.Infrastructure/Repositories/RestaurantRepository.cs:26]: use .ThenInclude() instead of second include()

Comment (RomanMykushLnu) [reply API/RestMatch.API/RestMatch.API.Infrastructure/Repositories/RestaurantRepository.cs:26]: not needed

File: API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs (hunk 4 above)
Comment (RomanMykushLnu) [reply API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs]: fixed

File: API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs (hunk 3 above)
Comment (RomanMykushLnu) [reply API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs]: added

File: API/RestMatch.API/RestMatch.API/Program.cs (hunk 2 above)
Comment (RomanMykushLnu) [reply API/RestMatch.API/RestMatch.API/Program.cs]: fixed

File: API/RestMatch.API/RestMatch.API/Program.cs (hunk 1 above)
Comment (RomanMykushLnu) [reply API/RestMatch.API/RestMatch.API/Program.cs]: fixed

**Pull request merged**